cmd/syncthing/main.go

```diff
@@ -770,21 +770,8 @@ func syncthingMain(runtimeOptions RuntimeOptions) {
 		miscDB.PutString("prevVersion", Version)
 	}
 
-	if cfg.RawCopy().OriginalVersion < 19 {
-		// Converts old symlink types to new in the entire database.
-		ldb.ConvertSymlinkTypes()
-	}
-	if cfg.RawCopy().OriginalVersion < 26 {
-		// Adds invalid (ignored) files to global list of files
-		changed := 0
-		for folderID, folderCfg := range folders {
-			changed += ldb.AddInvalidToGlobal([]byte(folderID), protocol.LocalDeviceID[:])
-			for _, deviceCfg := range folderCfg.Devices {
-				changed += ldb.AddInvalidToGlobal([]byte(folderID), deviceCfg.DeviceID[:])
-			}
-		}
-		l.Infof("Database update: Added %d ignored files to the global list", changed)
-	}
+	// Potential database transitions
+	ldb.Update(cfg.RawCopy().OriginalVersion)
```

Comment:
I'd just have a `const databaseSchemaVersion = 1` at the top of main.go and use that here. We can apply all the old transformations (again) when going 0 -> 1 and then bump it as appropriate in the future?

Reply:
Makes sense, I'll do that. There's just one thing: The global invalid transition isn't idempotent. At the time there were no global invalid entries, so the transition doesn't check whether there is already an entry present, it just inserts - that was a bad (fair-weather) idea... So i'll have to fix that too.